### ISSUE TITLE
forc-deploy cli tests should not coincide with local wallet instances while running cargo test locally

### ISSUE THREAD

**Issue by kayagokalp** [FuelLabs/sway#5484]
TL;DR: We need to make sure wallet instantiation is done at some tmp folder and cleared after the test.

related to #5482. Since tests are trying to deploy a contract, forc-deploy tries to look for local wallet in the default path. This is problematic because if a sway repo contributor has a forc-wallet installed at that path, the tests are stopped and vault password is asked to them. This caused me to miss couple of issues in the test suite because I thought the tests were passing. Instead tests were hanging as they were asking for my local forc-wallet password. This problem do not occur in CI environment as the environment is fresh and do not have any wallet installed. So rather than trying to unlock existing wallet, forc-deploy creates a new one in the default path which is killed before next run.

**Comment by kayagokalp** [comment]
Closed in favor of #6283
